aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -19,20 +19,82 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/cuda/CUDAStream.h>
 
+// TODO(NS): Investigate why FP8 conversion intrisncs end up being slower
+#if 0
+#include <cuda_fp8.h>
+#endif
+
 namespace at::native {
 
 void neg_kernel_cuda(TensorIteratorBase &iter);
 void conj_kernel_cuda(TensorIteratorBase &iter);
 
+void float8_copy_kernel_cuda(TensorIteratorBase &iter) {
+  ScalarType dtype = iter.dtype(0);
+  ScalarType other_dtype = iter.dtype(1);
+  if (dtype == kFloat8_e4m3fn) {
+    if (other_dtype == kFloat) {
```

Comment:
nit: would switch make this easier to read?

Reply:
Can you please elaborate what do you mean? You have top-level ifs for casts from float, bfloat16 and float16?